### ISSUE TITLE
False negative on the `Style/NumericPredicate` cop

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12433]
## Describe

```ruby
def my_method
  # bad
  'foo' if value&.> 0
end
```

## RuboCop

```console
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop -V
1.58.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 2.7.8) [x86_64-linux]
  - rubocop-capybara 2.19.0
  - rubocop-factory_bot 2.24.0
  - rubocop-performance 1.19.1
  - rubocop-rails 2.22.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.25.0
  - rubocop-thread_safety 0.5.1
```

**Comment by ydakuka** [comment]
I'm not sure if it should be done for this cop, since there are plenty of such constructions.
```ruby
hsh&.[](:foo)
arr&.[](1)&.to_i
```

**Comment by koic** [comment]
It's a very corner case, and it's unclear whether it's worth handling.
